### PR TITLE
[B2BP-116] Initialized bare Strapi CMS instance

### PR DESCRIPTION
**Pull request by simosalsi**
Added a bare instance of Strapi CMS. Depends on #30 .

This will serve as a base to be expanded on for the b2b-portals backend side: the admin panels that each product team will access to manage their respective showcase website.

#### List of Changes
- Initialized Strapi
- Configured TypeScrypt
- Cleaned up files
- Generated .gitignore through [TopTal gitignore.io tool](https://www.toptal.com/developers/gitignore/)
- Added .eslintrc.json from [this PagoPA repo](https://github.com/pagopa/developer-portal/blob/c1c5047/packages/gitbook-docs/.eslintrc.json)

#### Motivation and Context
First step in the deployment of a functioning CMS.

#### How Has This Been Tested?
Tested locally following the procedures detailed in README.md.

#### Types of changes
- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/strapi-cms/.env.example [hunk 1]
```diff
@@ -0,0 +1,6 @@
+HOST=0.0.0.0
+PORT=1337
+APP_KEYS=ErARKTh5IBFLmuOYZ+M5RA==,Ve3wVVMijnOpZjPKeuuYpA==,SewgPjTM+79/95IlQZoSUg==,ssoUQ6KLFdl8AymPl7FnQg==
+API_TOKEN_SALT=bKJsi30m4bF5APP5jVslkQ==
+ADMIN_JWT_SECRET=XqKNcnZsgkSkjBUeaK1mlw==
+TRANSFER_TOKEN_SALT=Wxa1v2/2taWKl+sT3paQQw==
```
Comment (datalek) [inline apps/strapi-cms/.env.example:6]:
```suggestion
HOST=0.0.0.0
PORT=1337
APP_KEYS=changeme1,changeme2
API_TOKEN_SALT=changeme
ADMIN_JWT_SECRET=changeme
TRANSFER_TOKEN_SALT=changeme
JWT_SECRET=changeme
```

File: apps/strapi-cms/README.md
Comment (datalek) [inline apps/strapi-cms/README.md:1]: Remove the information that are already into the root [README](https://github.com/pagopa/b2b-portals/blob/main/README.md)

File: apps/strapi-cms/package.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "strapi-cms",
+  "private": true,
+  "version": "0.1.0",
```
Comment (datalek) [inline apps/strapi-cms/package.json:4]:
```suggestion
  "version": "0.0.0",
```

File: apps/strapi-cms/package.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  "scripts": {
+    "dev": "strapi develop",
+    "start": "strapi start",
+    "build": "strapi build",
+    "strapi": "strapi"
```
Comment (datalek) [inline apps/strapi-cms/package.json:9]:
```suggestion
    "clean": "shx rm -rf dist/",
    "compile": "npm run clean && tsc --noEmit",
    "dev": "strapi develop",
    "start": "strapi start",
    "build": "strapi build",
    "lint": "eslint --ignore-path .gitignore src",
    "strapi": "strapi"
```

File: apps/strapi-cms/package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  "devDependencies": {},
```
Comment (datalek) [inline apps/strapi-cms/package.json:11]:
```suggestion
  "devDependencies": {
    "eslint": "^8.50.0",
    "typescript": "5.2.2"
  },
```

File: apps/strapi-cms/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
+{
+  // "extends": "@strapi/typescript-utils/tsconfigs/server",
```
Comment (datalek) [inline apps/strapi-cms/tsconfig.json:2]:
```suggestion
```

File: apps/strapi-cms/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  "extends": [
+    "@strapi/typescript-utils/tsconfigs/server",
+    "../../tsconfig"
+  ],
+  "compilerOptions": {
+    "outDir": "dist",
+    "rootDir": ".",
+    // From "@strapi/typescript-utils/tsconfigs/server",
```
Comment (datalek) [inline apps/strapi-cms/tsconfig.json:10]:
```suggestion
```

File: apps/strapi-cms/.eslintrc.json [hunk 7]
```diff
@@ -0,0 +1,6 @@
+{
+    "root": true,
+    "extends": [
+      "custom"
```
Comment (datalek) [inline apps/strapi-cms/.eslintrc.json:4]: To make this `custom` work we have to define also the `eslint-config` package. I'm going to add it on a separate PR.

**Review by datalek — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/package.json [hunk 8]
```diff
@@ -23,6 +23,7 @@
     "eslint-config-next": "13.4.19",
     "eslint-config-prettier": "^9.0.0",
     "eslint-plugin-prettier": "^5.0.0",
-    "typescript": "5.2.2"
+    "typescript": "5.2.2",
+    "synckit": "^0.8.5"
```
Comment (kin0992) [inline apps/nextjs-website/package.json:27]: what do you need this dependency for? is it really needed based on the scope of this activity?

Comment (simosalsi) [reply apps/nextjs-website/package.json:27]: Commit wouldn't work without it.

Comment (kin0992) [reply apps/nextjs-website/package.json:27]: What error do you have on commit if you remove `synckit`?

Comment (simosalsi) [reply apps/nextjs-website/package.json]: None of course. It fixed itself. It had something to do with lint so I would assume setting the configuration fixed the problem.

Already committed the change, thank you!

**Review by datalek — CHANGES_REQUESTED** [review]

File: apps/strapi-cms/README.md
[the remark this answers is not in the document]
Comment (simosalsi) [reply apps/strapi-cms/README.md:1]: I'm not against moving the steps needed for running the instance to the root README.

I think completely removing everything else would be reductive for someone who has to maintain the code though.

Running it is one thing, but I think the information about the Content Type Builder, the necessity for `npm run build` and the steps Strapi takes when launching `npm run dev` could save some headaches to devs that aren't familiar with Strapi.

File: apps/strapi-cms/types/generated/components.d.ts
Comment (datalek) [inline apps/strapi-cms/types/generated/components.d.ts:1]: Is this file required now? I tried to remove it and the cms works anyway

File: apps/strapi-cms/types/generated/contentTypes.d.ts
Comment (datalek) [inline apps/strapi-cms/types/generated/contentTypes.d.ts:1]: Same for this file. It is required now?

**Review by datalek — CHANGES_REQUESTED** [review]

File: apps/strapi-cms/types/generated/contentTypes.d.ts
Comment (simosalsi) [reply apps/strapi-cms/types/generated/contentTypes.d.ts:1]: The entire generated folder can be omitted for now, you're right.

I'm going to delete it manually and commit again by EOD if I'm not instructed otherwise.

I could update the .gitignore but since it's been generated by TopTal I trust there is a reason for which they've not excluded the generated folder, so I'm not going to modify it.

**Review by datalek — APPROVED** [review]
📣
